api/filter/http/gzip.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+// [#proto-status: experimental]
+
+package envoy.api.v2.filter.http;
+
+import "validate/validate.proto";
+
+// This is an HTTP filter which enables Envoy to compress dispatched data from an upstream
+// service upon client request. This is useful in situations where large payloads needs to
+// be transmitted without compromising the response time. Note that when this filter is
+// activated, content will be served with chunked transfer-encoding and gzip content-encoding.
```

Comment:
I think this statement only applies if the response is actually compressed per the rules in the configuration, right?